htlcswitch/hop/payload.go

```diff
@@ -54,8 +54,57 @@ func (v PayloadViolation) String() string {
 	}
 }
 
+// RouteRole represents the different types of roles a node can have as a
+// recipient of a HTLC.
+type RouteRole uint8
+
+const (
+	// RouteRoleCleartext represents a regular route hop.
+	RouteRoleCleartext RouteRole = iota
+
+	// RouteRoleIntroduction represents an introduction node in a blinded
+	// path, characterized by a blinding point in the onion payload.
+	RouteRoleIntroduction
```

Comment:
> or will this be important for receiving?

Once we add receiving, we'll want to know if we're the intro node and the recipient because we _don't_ blind the error then. 

Also think it's just nice for logging in this PR?

Reply:
yeah +1 for the logging 👍

Reply:
> Once we add receiving, we'll want to know if we're the intro node and the recipient because we don't blind the error then.



Why do we only care about if we're the intro node when we add receving? IIUC, even as is, we can be the introductory node, so therefore need to still special case the way we handle an error _received_ from the tunnel.

Reply:
> Why do we only care about if we're the intro node when we add receving? I

We always care about it, just in this approach we don't actually _need_ the `RouteRoleIntroduction` because we had the payload blinding point on hand. Once we get to handling errors where we're the recieiving node in a blinded route, we don't have that blinding point on hand so we'll use this enum variant in those cases.